sklearn/utils/estimator_checks.py

```diff
@@ -1345,12 +1345,12 @@ def check_dtype_object(name, estimator_orig):
     if hasattr(estimator, "transform"):
         estimator.transform(X)
 
-    with raises(Exception, match="Unknown label type", may_pass=True):
+    with raises(Exception, match="(?i)unknown", may_pass=True):
         estimator.fit(X, y.astype(object))
 
     if "string" not in tags["X_types"]:
         X[0, 0] = {"foo": "bar"}
-        msg = "argument must be a string.* number"
+        msg = "string.* number"
```

Comment:
here as well, we're basically loosening up the tests here, I rather fix the estimator.

Reply:
I revert back with a slight change in the `match`. I added a comment to inform from where the error is raised.